brozzler/worker.py

```diff
@@ -499,11 +502,23 @@ def brozzle_site(self, browser, site):
                 # using brozzler-worker --proxy, nothing to do but try the
                 # same proxy again next time
                 logging.error("proxy error (self._proxy=%r)", self._proxy, exc_info=1)
-        except:
-            self.logger.error(
-                "unexpected exception site=%r page=%r", site, page, exc_info=True
-            )
+        except (brozzler.PageConnectionError, Exception) as e:
+            if isinstance(e, brozzler.PageConnectionError):
+                self.logger.error(
+                    "Page status code possibly indicates connection failure between host and warcprox: site=%r page=%r",
+                    site,
+                    page,
+                    exc_info=True,
+                )
+            else:
+                self.logger.error(
+                    "unexpected exception site=%r page=%r", site, page, exc_info=True
+                )
             if page:
+                retry_delay = min(60, 60 * (1.5**page.failed_attempts))
```

Comment:
This `min()` should definitely be `max()` and also needs a comment explaining that this is a simple backoff calculation yielding 60, 90, 135 for attempts of 0, 1, 2